packages/browser-integration-tests/suites/replay/bufferMode/test.ts

```diff
@@ -0,0 +1,422 @@
+    await page.click('#go-background');
+
+    // Has stopped recording, should make no more calls to Sentry
+    expect(callsToSentry).toEqual(3);
+
+    const event0 = getReplayEvent(req0);
+    const content0 = getReplayRecordingContent(req0);
+
+    expect(event0).toEqual(
+      getExpectedReplayEvent({
+        contexts: { replay: { error_sample_rate: 0, session_sample_rate: 0 } },
+        error_ids: [errorEventId!],
+        replay_type: 'buffer',
+      }),
+    );
+
+    // The first event should have both, full and incremental snapshots,
+    // as we recorded and kept all events in the buffer
+    expect(content0.fullSnapshots).toHaveLength(1);
+    // We don't know how many incremental snapshots we'll have (also browser-dependent),
+    // but we know that we have at least 5
+    expect(content0.incrementalSnapshots.length).toBeGreaterThan(5);
+    // We want to make sure that the event that triggered the error was recorded.
+    expect(content0.breadcrumbs).toEqual(
+      expect.arrayContaining([
+        {
+          ...expectedClickBreadcrumb,
+          message: 'body > button#error2',
+          data: {
+            nodeId: expect.any(Number),
+            node: {
+              attributes: {
+                id: 'error2',
+              },
+              id: expect.any(Number),
+              tagName: 'button',
+              textContent: '******* *****',
+            },
+          },
+        },
+      ]),
+    );
+  },
+);
+
+// Doing this in buffer mode to test changing error sample rate after first
+// error happens.
+sentryTest('[buffer-mode] can sample on each error event', async ({ getLocalTestPath, page, browserName }) => {
+  // This was sometimes flaky on firefox/webkit, so skipping for now
+  if (shouldSkipReplayTest() || ['firefox', 'webkit'].includes(browserName)) {
+    sentryTest.skip();
+  }
+
+  let callsToSentry = 0;
+  const errorEventIds: string[] = [];
+  const reqPromise0 = waitForReplayRequest(page, 0);
+  const reqErrorPromise = waitForErrorRequest(page);
+
+  await page.route('https://dsn.ingest.sentry.io/**/*', route => {
+    const event = envelopeRequestParser(route.request());
+    // error events have no type field
+    if (event && !event.type && event.event_id) {
+      errorEventIds.push(event.event_id);
+    }
+    // We only want to count errors & replays here
+    if (event && (!event.type || isReplayEvent(event))) {
+      callsToSentry++;
+    }
+
+    return route.fulfill({
+      status: 200,
+      contentType: 'application/json',
+      body: JSON.stringify({ id: 'test-id' }),
+    });
+  });
+
+  const url = await getLocalTestPath({ testDir: __dirname });
+
+  await page.goto(url);
+  // Start buffering and assert that it is enabled
+  expect(
+    await page.evaluate(() => {
+      const replayIntegration = (window as unknown as Window & { Replay: InstanceType<typeof Replay> }).Replay;
+      // @ts-ignore private
+      const replay = replayIntegration._replay;
```

Comment:
```suggestion
      const replay = replayIntegration['_replay'];
```

I think this way we do not need to ts-ignore?